docs/pip_sphinxext.py

```diff
@@ -13,6 +15,41 @@
 from pip._internal.req.req_file import SUPPORTED_OPTIONS
 
 
+class PipTowncrierDraft(rst.Directive):
+    required_arguments = 0
+
+    def _find_section_title(self, lines):
+        for i, line in enumerate(lines):
+            if not re.match(r"^=+$", line) or i == 0:
+                continue
+            title_match = re.match(
+                r"^(?P<version>[^ ]+) \((?P<date>[^\)]+)\)$",
+                lines[i - 1],
+            )
+            if not title_match:
+                continue
+            return i - 1, title_match.group("version", "date")
+        return None, (None, None)
+
+    def run(self):
+        lines = subprocess.check_output(
+            [sys.executable, "-m", "towncrier", "--draft"],
+            stderr=subprocess.DEVNULL,
+            text=True,
+        ).splitlines(keepends=False)
+
+        index, (version, date) = self._find_section_title(lines)
+        assert index is not None, "section title not found"
+        assert version is not None, "version not found in title"
+        assert date is not None, "date not found in title"
+
+        lines[index] = f"{version}, unreleased as in {date}"
```

Comment:
I'd just change this to "unreleased, as on {date}". Also, yay f-strings!

Reply:
It should rather be "as **of** {date}"

Reply:
> I'd just change this to "unreleased, as on {date}"

Do you mean dropping the `{version}` part

Reply:
Drop the version basically. IDC about grammar here. 🤷🏽